exec/vector/src/main/codegen/templates/BasicTypeHelper.java

```diff
@@ -233,10 +227,54 @@ public static ValueVector getNewVector(String name, BufferAllocator allocator, M
       throw new UnsupportedOperationException(buildErrorMessage("get holder reader implementation", type, mode));
   }
   
-  public static ValueVector getNewVector(MaterializedField field, BufferAllocator allocator){
+  public static ValueVector getNewVector(String name, BufferAllocator allocator, MajorType type, CallBack callBack) {
+    MaterializedField field = MaterializedField.create(name, type);
+    return getNewVector(field, allocator, callBack);
+  }  
+  
+  public static ValueVector getNewVector(MaterializedField field, BufferAllocator allocator) {
```

Comment:
`getNewVector(MaterializedField field, ...` promotes `field` sharing between vectors and possibly threads. What are the use cases where `MaterializedField` needs to be shared between vectors? I mean, when is it necessary that modifications to `MaterializedField` that belongs to one vector are reflected transparently in another vector `MaterializedField`?

Reply:
No sharing should ever occur. Vectors under construction are not sharable; only completed, finalized vectors are shareable. (None of the vector builder methods are synchronized. Not sure what it even means for two threads to write to the same vector as they must coordinate on rows, and thread T1 can't write its VARCHAR to row 2 until thread T2 writes its value to row 1.)

Reply:
This is a red herring. This file is never used. Please see the correct fix in `PartionSenderTemplate`.

Reply:
I perhaps misunderstood the original comment.

There are two ways that `getNewVector` is used:

* The caller builds up a `MaterializedField` and passes it to the new vector. No sharing occurs. This is the normal case used by, as far as I can tell, all but one place in the code.
* The `PartitionSenderTemplate` creates a new vector using the type of a previous vector. Here, the `MaterializedField` is already in use and must be cloned.

There seems no good reason to clone every `MaterializedField` since most uses are correct. In fact, the new result set loader code rather counts on the fact that the passed-in `MaterializeField` is the one that the vector uses.

We could possibly insert code into the `getNewVector()` that triggers an assert if the `MaterializedField` already has children (or its minor type has subtypes). That will catch field sharing if it occurs again.

Doing so might uncover other improper usages, so is a bit out of the scope of this PR. Good exercise for someone who has access to the full pre-commit test suite.